app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/BrowsePageView.kt

```diff
@@ -373,23 +399,30 @@ class BrowsePageView(val state: RHMIState, val musicImageIDs: MusicImageIDs, val
 				if (item == SEARCHRESULT_EMPTY || item == SEARCHRESULT_SEARCHING) {
 					// invalid selection, don't change states
 					inputComponent.getSuggestAction()?.asHMIAction()?.getTargetModel()?.asRaIntModel()?.value = 0
+
 					throw RHMIActionAbort()
 				} else if (item == BrowseView.SEARCHRESULT_PLAY_FROM_SEARCH) {
 					browseController.playFromSearch(this.input)
 					browseController.onListSelection(item, inputComponent.getSuggestAction()?.asHMIAction())
 				} else {
-					previouslySelected = item  // update the selection state for future redraws
+					// update the selection state for future redraws
+					previouslySelected = item
+
 					browseController.onListSelection(item, inputComponent.getSuggestAction()?.asHMIAction())
 				}
 			}
 
 			override fun convertRow(row: MusicMetadata): String {
 				return if (row.subtitle != null) {
-					UnicodeCleaner.clean("${row.title}\n${row.subtitle}")
+					UnicodeCleaner.clean("${row.title} - ${row.subtitle} - ${row.artist}")
 				} else {
 					UnicodeCleaner.clean(row.title ?: "")
 				}
 			}
+
+			override fun onOk() {
+				browseController.showSearchResults(deferredSearchResults, inputComponent.getResultAction()?.asHMIAction())
```

Comment:
Perhaps this could be conditional? I could imagine the scenario where the deferredSearchResults are complete but empty, and I could imagine the desired `onOk` behavior in this situation would be to `PlayFromSearch` if available.

Reply:
Yeah that's a good idea, updated this so now when the search results come back empty and Spotify is not being used the PlayFromSearch method will be called with the query input